### ISSUE TITLE
How to perform FMR frequency excitation characterization?

### ISSUE THREAD

**Issue by catrevil** [ubermag/workshop#17]
Hello!

Thank you for these workshops.

Ubermag is great and very intuitive. It seems to me there must be a nice concise way to use ubermag to perform an fmr frequency-swept (or pulse or related methods) excitation characterization of a material.

Can you comment on how something like this might be set-up & performed using ubermag? (With either oommf or mumax as the underlying solver)?

Best,

CA

**Comment by marijanbeg** [comment]
Hi @catrevil, thank you for your question (and for helping with other raised issues - much appreciated!). There are a few ways how you can simulate the FMR response. Could you please give me the description of steps you would like to carry out and I will try to make a short tutorial about that.

**Comment by catrevil** [comment]
Hi @marijanbeg ! I am hoping to be able to characterize the resonant eigenmodes of some material in some geometry which is under a bias magnetisation. It is my understanding that some gaussian or sinc pulse would be the easiest way to excite all modes of a system and from that obtain the full resonant frequency spectrum.

I am hoping to program some sort of pulse profile or excitation procedure of some sort (i.e. pulsed excitation versus a frequency-swept excitation), and from that obtain the information needed for a frequency spectrum. Then, it seems to me, that obtaining the eigenmodes after this would be trivial, but it might be nice to realize some sort of peak-finder, in a case like that.

I think I have seen notes that for a generic material & geometry, a hysteresis measurement would be necessary prior to trying to obtain the fmr spectrum--I think you were not able to cover this in the time allotted in the last lesson, but I imagine defining a sequenced procedure for purposes such as those may be similar to one another & valuable to show.

I hope this makes sense, thank you for your help!

**Comment by marijanbeg** [comment]
Hi @catrevil, thank you for the clarification. Give me some time and I will make a tutorial about some elements you might need to do what you want. I will not be able to cover that during the next live session, but we might make a YouTube video.

**Comment by marijanbeg** [comment]
Hi @catrevil, is the "ringdown" method (second half of the methods section https://journals.aps.org/prb/abstract/10.1103/PhysRevB.95.014433) what you refer to? Getting the resonance frequencies (exciting the system with a sinc pulse) and then extracting the magnetisation dynamics associated to peaks (by filtering or driving the system using sine wave)?

**Comment by catrevil** [comment]
@marijanbeg yes--that is precisely it! What a great paper too, thank you for the reference.

I would very much appreciate seeing how that process could be streamlined using ubermag. By extracting the magnetisation dynamics associated with the peaks, this would mean recovering the resonant eigenmodes?

I am in the process of trying to write a generalized mumax3 script generator in python for the purposes of running this fmr pulsed signal excitation routine & possibly recovering the eigenmodes as well, but I think having this process laid out in ubermag would be immensely helpful & provide a greater level of functionality in the end.

**Comment by PkRPintu** [comment]
Dear @marijanbeg  and @catrevil ,

Hope you are doing well. I would also like to do a simulation of ferromagnetic resonance analysis using ubermag. I want to study the resonant eigenmodes in both field sweep and frequency sweep modes by exciting the system with a high-frequency (RF) field. If you guys come up with the tutorials or any other paper which will help in this regard then please let me know. Thanks a lot for your help.

Best wishes
Pradeep

**Comment by marijanbeg** [comment]
Hi @PkRPintu and @catrevil, sorry about the delay addressing this issue. It is on my todo list to make a tutorial about eigenmodes, but I have several urgent things I could not deal with during the workshop and that keeps me busy at the moment. So, I ask you for a bit more patience.

Thank you for your understanding.

**Comment by SukantaKumarJena** [comment]
Hello @marijanbeg  Thank you for the online session. I am also doing FMR spectra simulation. I would like to add iDMI term and simulate field vs frequency. Is it possible to simulate from zero field to Higher field and with some range of frequency (ex-2-10GHz) to get the dispersion plot? Cold you please give some suggestions how to proceed further?

Thanks,
Sukanta Kumar Jena
